oryx-app-mllib/src/main/java/com/cloudera/oryx/app/mllib/kmeans/KMeansEvaluation.java

```diff
@@ -126,6 +127,38 @@
     return dunnIndex;
   }
 
+  /**
+   * Computes the Silhouette Coefficient for this clustering, range is [-1,1]
+   * (http://www.sciencedirect.com/science/article/pii/0377042787901257)
+   * @param evalData data for evaluation
+   * @return Silhouette Coefficient, closer to 1 is better
+   */
+  double silhouetteCoefficient(JavaRDD<Vector> evalData) {
+    Map<Integer, List<double[]>> clusteredPointsMap =
```

Comment:
Hm, I'm concerned about this. The entire data set has to fit in memory in the driver here then.